nautobot/extras/forms/forms.py

```diff
@@ -1235,6 +1241,72 @@ class JobButtonFilterForm(BootstrapMixin, forms.Form):
     )
 
 
+#
+# Metadata
+#
+
+
+# MetadataChoice inline formset for use with providing dynamic rows when creating/editing choices
+# for `MetadataType` objects in UI views. Fields/exclude must be set but since we're using all the
+# fields we're just setting `exclude=()` here.
+MetadataChoiceFormSet = inlineformset_factory(
+    parent_model=MetadataType,
+    model=MetadataChoice,
+    exclude=(),
+    extra=5,
+    widgets={
+        "value": forms.TextInput(attrs={"class": "form-control"}),
+        "weight": forms.NumberInput(attrs={"class": "form-control"}),
+    },
+)
+
+
+class MetadataTypeForm(NautobotModelForm):
+    name = forms.CharField(required=True, max_length=CHARFIELD_MAX_LENGTH)
+    description = forms.CharField(required=False, max_length=CHARFIELD_MAX_LENGTH)
+    content_types = MultipleContentTypeField(
+        feature="metadata", help_text="The object(s) to which Metadata of this type can be applied."
+    )
+
+    class Meta:
+        model = MetadataType
+        fields = (
+            "name",
+            "description",
+            "data_type",
+            "content_types",
+            "tags",
+        )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.instance and self.instance.present_in_database:
+            self.fields["data_type"].widget.attrs["readonly"] = True
```

Comment:
This doesn't seem to be working properly for me. The widget is greyed out but I can still click on it and change the value.

Reply:
Interesting. This was a copy/paste from the CustomField form, so it's probably an issue there too. I'll look into it.